.azure-pipelines/main-pipeline.yml

```diff
@@ -80,65 +79,70 @@ stages:
+                  connectTo: 'VsTeam'
+                  connectedServiceName: 'marketplace-service-connection'
+                  publisherId: '$(publisherId)'
+                  extensionId: '$(extensionId)'
+                  extensionTag: '$(devExtensionTag)'
+                  versionAction: 'Patch'
 
-          # We need to wait in this job (i.e. inside the main pipeline) to make sure the extension is available before the E2E pipeline's jobs are enqueued.
-          - task: Bash@3
-            displayName: Wait for DEV extension availability
-            inputs:
-              filePath: 'ci/poll-dev-extension-availability.sh'
-              arguments: |
-                -s "$(SERVICE_URL)" \
-                -t "$(SERVICE_TOKEN)" \
-                -i "$(publisherId).$(extensionId)$(devExtensionTag).$(contributionId)" \
-                -v "$(queryDevVersion.Extension.Version)"
-
-  # Extension public release, only performed on manual runs
-  - ${{ if eq(parameters.performRelease, true) }}:
+              - task: PublishAzureDevOpsExtension@4
+                displayName: Publish new DEV extension version
+                inputs:
+                  connectTo: 'VsTeam'
+                  connectedServiceName: 'marketplace-service-connection'
+                  fileType: 'vsix'
+                  vsixFile: '$(Pipeline.Workspace)/$(artifactPath)/$(packagedTask)'
+                  extensionTag: '$(devExtensionTag)'
+                  extensionName: '$(extensionName) DEV'
+                  extensionVisibility: 'private'
+                  extensionPricing: 'free'
+                  shareWith: 'datadog-ci'
+                  extensionVersion: '$(queryDevVersion.Extension.Version)'
+                  updateTasksId: true # The DEV task ID is overridden to `99cf84cd-0117-58b7-b1dd-d95e5ae9baf8` (based on the `extensionTag`)
+                  updateTasksVersion: true # Do not require updating the task version during development
+                  updateTasksVersionType: patch
+
+              # We need to wait in this job (i.e. inside the main pipeline) to make sure the extension is available before the E2E pipeline's jobs are enqueued.
+              - task: Bash@3
+                displayName: Wait for DEV extension availability
+                inputs:
+                  filePath: 'ci/poll-dev-extension-availability.sh'
+                  arguments: |
+                    -s "$(SERVICE_URL)" \
+                    -t "$(SERVICE_TOKEN)" \
+                    -i "$(publisherId).$(extensionId)$(devExtensionTag).$(contributionId)" \
+                    -v "$(queryDevVersion.Extension.Version)"
+
+  # Extension public release, only performed on release tags
+  - ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/v') }}:
       - stage: ReleasePublic
         displayName: Release public extension
+        dependsOn: Build
+        variables:
+          extensionVersion: $[replace(variables['Build.SourceBranch'], 'refs/tags/v', '')]
```

Comment:
Instead of using `QueryAzureDevOpsExtensionVersion@4` to query (and apply the semver change to) the current version of the extension, we get the version based on the tag name.